adapter/list/config/config.proto

```diff
@@ -0,0 +1,68 @@
+// Copyright 2016 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package adapter.ipListChecker.config;
+
+import "google/protobuf/duration.proto";
+import "gogoproto/gogo.proto";
+
+option go_package="config";
+option (gogoproto.goproto_getters_all) = false;
+option (gogoproto.equal_all) = false;
+option (gogoproto.gostring_all) = false;
+
+message Params {
+    // Where to find the list to check against. This may be ommited for a completely local list.
+    string provider_url = 1;
+
+    // Determines how often the provider is polled for
+    // an updated list
+    google.protobuf.Duration refresh_interval = 2 [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+
+    // Indicates how long to keep a list before discarding it.
+    // Typically, the TTL value should be set to noticeably longer (> 2x) than the
+    // refresh interval to ensure continued operation in the face of transient
+    // server outages.
+    google.protobuf.Duration ttl = 3 [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+
+    // Indicates the amount of time a caller of this adapter can cache an answer
+    // before it should ask the adapter again.
+    google.protobuf.Duration caching_interval = 4 [(gogoproto.nullable) = false, (gogoproto.stdduration) = true];
+
+    // Indicates the number of times a caller of this adapter can use a cached answer
+    // before it should ask the adapter again.
+    int64 caching_use_count = 5;
+
+    // List entries that are consulted first, before the list from the server
+    repeated string overrides = 6;
+
+    enum ListEntryType {
```

Comment:
We can consider it, in a subsequent PR if you don't mind.

I'm not thrilled with the support for IP_ADDRESS as it is. The adapter is consuming a string that it treats as an IP address, instead of consuming a value of type "IP address" which would type check against an attribute of that type. I was thinking about adding an IPListEntry template for this purpose and have this adapter consume IP addresses as either strings or actual IP addresses.